File: kshitij/ESE/euclidean.cpp
// Euclidean and Extended C++

#include <bits/stdc++.h>
using namespace std;

// Eucldiean
int gcd(int a, int b) {
    if (b == 0)return a;
    return gcd(b, a % b);
}

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int q, r1, r2, r, s1 = 1, s2 = 0, s, t1 = 0, t2 = 1, t;
        cin >> r1 >> r2;
        cout << r1 << " " << r2 << "\n";
        cout << gcd(r1, r2) << "\n";
        cout << "q r1 r2 r s1 s2 s t1 t2 t\n";
        // Extended Euclidean
        while (r2 > 0) {
            cout << q << " " << r1 << " " << r2 << " " << r << " " << s1 << " " << s2 << " " << t1 << " " << t2 << "\n";
            q = r1 / r2;
            r = r1 - r2 * q;
            r1 = r2;
            r2 = r;
            s = s1 - s2 * q;
            s1 = s2;
            s2 = s;
            t = t1 - t2 * q;
            t1 = t2;
            t2 = t;
        }
        cout << "Gcd: " << r1 << "\n s: " << s1 << "\n t: " << t1 << "\n";

    }
    return 0;
}

File: Sweety/euclidean.py
def euclidean(a: int, b: int):
  r1 = a
  r2 = b

  print(f'{"q"} {"r1":>20} {"r2":>20} {"r":>20}')
  print("")

  while r2 != 0:
    q = r1 // r2
    r = r1 - q * r2
    print(f'{q} {"%20.0d" % r1} {"%20.0d" % r2} {"%20.0d" % r}')
    
    r1 = r2
    r2 = r
  
  print(f'{"-"} {"%20.0d" % r1} {"%20.0d" % r2} {"-":>20}')
  
  print(f'GCD({a}, {b}) = {r1}')
  



def main():

  a = int(input('Enter a: '))
  b = int(input('Enter b: '))
  euclidean(a,b)


   

main()  
  
